Dev

vvverman opened the pull request.

vvverman committed:
15 спринт

Разложил файлы по пакетам Data, Presentation, Domain. Создал файл с интерфейсами для работы с треками.

vvverman committed:
15 спринт

Разложил файлы по пакетам Data, Presentation, Domain. Создал файл с интерфейсами для работы с треками. Добавил комментарии

vvverman committed:
15 спринт

Пытаюсь создать интерфейс для работы с SharedPreferences

vvverman committed:
15 спринт

Вроде разложил, чо нашёл...

vvverman committed:
15 спринт

Вроде разложил, чо нашёл...

vvverman committed:
15 спринт

убрал hashcode и equals

Charmandik commented on app/src/main/java/com/example/playlistmaker/data/SettingsManager.kt:
```diff
@@ -0,0 +1,29 @@
+}
+
+
+class SettingsManagerImpl(private val sharedPreferences: SharedPreferences) : SettingsManager {
```
старайся разносить разные классы или интерфейсы по разным файлам. Один класс/интерфейс в один файл. Это позволяет гораздо легче ориентироваться в структуре проекта и упрощает чтение, так как ты заходишь в нужный тебе класс и читаешь только то, чем занимается класс

Charmandik commented on app/src/main/java/com/example/playlistmaker/presentation/MainActivity.kt:
```diff
@@ -0,0 +1,48 @@
+        // Находим кнопку по ее идентификатору
+        val searchButton = findViewById<Button>(R.id.editTextSearch)
+
+        // Устанавливаем слушатель клика на кнопку поиска
```
если приходится для самого себя объяснять - что же здесь происходит, то тогда самым простым вариантом можем стать вынесением в функцию. Например, этот можно было бы вынести в функцию setListenerToSearchButton и из названия функции нам уже понятно, что эта функция устанавливает слушатель на кнопку поиска и нам не нужно писать комментарий

Charmandik commented on app/src/main/java/com/example/playlistmaker/presentation/MediaLibraryActivity.kt:
```diff
@@ -13,8 +13,17 @@ import com.bumptech.glide.Glide
 class MediaLibraryActivity : AppCompatActivity() {
+
+    val trackService: TrackService = TrackServiceImpl()
+    private lateinit var settingsManager: SettingsManager
```
Вместо использования lateinit, я рекомендую тебе использовать подход с добавлением nullability к типу объекта с оператором ?. и сразу проинициализировать его значением null. Это решение позволяет избежать ошибок, связанных с попытками доступа к неинициализированным объектам, которые могут привести к непредсказуемому поведению или крашам приложения. Хотя это может добавить некоторую дополнительную работу при обращении к объекту через оператор ?. , но это значительно повысит безопасность приложения, так как ты будешь уверен, что объект всегда либо имеет значение, либо равен null, и не будет вызывать исключительных ситуаций.Можешь еще прочитать статью на эту тему https://habr.com/ru/companies/maxilect/articles/590061

Charmandik commented on app/src/main/java/com/example/playlistmaker/presentation/ThemeManager.kt:
```diff
@@ -0,0 +1,40 @@
+import android.app.Application
+import androidx.appcompat.app.AppCompatDelegate
+
+class ThemeManager : Application() {
```
Всё же это не только менеджер темы)
Это основной класс, в котором в дальнейшем будут цепляться различные библиотеки и события, поэтому стоит переименовать)

vvverman committed:
15 спринт

Перенёс
interface SettingsManager - в  domain
interface TrackService - в domain
SearchHistory - в data
